tests/integration/large_deployments/test_orchestration.py

```diff
@@ -0,0 +1,353 @@
+            _2nd_DATA_APP: {"active": []},
+        },
+        units_full_statuses={
+            MAIN_APP: {"units": {"active": []}},
+            FAILOVER_APP: {"units": {"active": []}},
+            DATA_APP: {"units": {"active": []}},
+            _2nd_DATA_APP: {"units": {"active": []}},
+        },
+        wait_for_exact_units=1,
+    )
+
+    # check that failover was promoted to main orchestrator
+    data_app = ops_test.model.applications[DATA_APP]
+    assert data_app is not None
+    orchestrators = await get_application_relation_data(
+        ops_test,
+        unit_name=data_app.units[0].name,
+        relation_name=PeerRelationName,
+        key="orchestrators",
+    )
+    assert orchestrators, "No orchestrators found in relation data"
+    orchestrators = PeerClusterOrchestrators.from_dict(json.loads(orchestrators))
+    assert (
+        orchestrators.main_app and orchestrators.main_app.name == FAILOVER_APP
+    ), "Failover was not promoted to main orchestrator"
+    assert (
+        orchestrators.failover_app is None
+    ), "Failover orchestrator should be None after promotion"
+
+
+@pytest.mark.abort_on_fail
+async def test_failover_election_after_relation_integration(ops_test: OpsTest) -> None:
+    """Test that the failover orchestrator is correctly elected after re-adding relations."""
+    assert ops_test.model is not None
+
+    await ops_test.model.integrate(
+        f"{FAILOVER_APP}:peer-cluster-orchestrator", f"{MAIN_APP}:peer-cluster"
+    )
+    for app in [DATA_APP, _2nd_DATA_APP]:
+        await ops_test.model.integrate(
+            f"{MAIN_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    await wait_until(
+        ops_test,
+        apps=[MAIN_APP, DATA_APP, _2nd_DATA_APP],
+        apps_statuses=["active"],
+        units_full_statuses={
+            MAIN_APP: {"units": {"active": []}},
+            DATA_APP: {"units": {"active": []}},
+            _2nd_DATA_APP: {"units": {"active": []}},
+        },
+        wait_for_exact_units=1,
+    )
+
+    # check that main app is now elected failover orchestrator
+    data_app = ops_test.model.applications[DATA_APP]
+    assert data_app is not None
+    orchestrators = await get_application_relation_data(
+        ops_test,
+        unit_name=data_app.units[0].name,
+        relation_name=PeerRelationName,
+        key="orchestrators",
+    )
+    assert orchestrators, "No orchestrators found in relation data"
+    orchestrators = PeerClusterOrchestrators.from_dict(json.loads(orchestrators))
+    assert (
+        orchestrators.main_app and orchestrators.main_app.name == FAILOVER_APP
+    ), "Failover is supposed to be the main orchestrator"
+    assert (
+        orchestrators.failover_app and orchestrators.failover_app.name == MAIN_APP
+    ), "Main app is supposed to be the failover orchestrator"
+
+
+@pytest.mark.abort_on_fail
+async def test_scale_failover_to_0_and_back(ops_test: OpsTest) -> None:
```

Comment:
I would not mention failover, as it is the current main orchestrator app.
```suggestion
async def test_scale_promoted_main_to_0_then_up(ops_test: OpsTest) -> None:
```